test/types/basics.test.ts

```diff
@@ -48,7 +48,20 @@ describe('=> Basics (TypeScript)', function() {
     summary: string;
 
     @Column(TEXT)
-    settings: string;
+    get settings(): Record<string, any> {
+      const text = this.attribute('settings') as string;
```

Comment:
technically Record<string, any> is still assignable to LIteral but somehow this works, this.attribute('settings') returns Literal here.

Reply:
might because of the extra `U extends T[Key]` that derives a new type from `T[Key]`?

Reply:
it seems `text`'s type is `Literal | this["settings"]`